### PR TITLE
feat: Adds new `integration_id` attribute in `mongodbatlas_alert_configuration`

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUD-228077

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: website/docs/r/alert_configuration.html.markdown [hunk 1]
```diff
@@ -119,6 +119,22 @@ resource "mongodbatlas_alert_configuration" "test" {
...
+
+```terraform
+resource "mongodbatlas_alert_configuration" "test" {
+  project_id = "PROJECT ID"
+  enabled    = true
+  event_type = "USERS_WITHOUT_MULTI_FACTOR_AUTH"
+
+  notification {
+    type_name     = "PAGER_DUTY"
+    integration_id = "THIRD PARTY INTEGRATION ID"
```
Comment (AgustinBettati) [inline website/docs/r/alert_configuration.html.markdown:133]: Our [third_party_integration data source](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/third_party_integration) currently is not capable of obtaining this id (while it is being returned in the API), will create a follow up ticket to address this.

File: internal/service/alertconfiguration/resource_alert_configuration_test.go [hunk 2]
```diff
@@ -421,6 +422,32 @@ func TestAccConfigRSAlertConfiguration_withPagerDuty(t *testing.T) {
 	})
 }
 
+func TestAccConfigAlertConfiguration_PagerDutyUsingIntegrationID(t *testing.T) {
+	acc.SkipTestForCI(t) // currently skipped as third party integration id cannot be created and obtained during test
```
Comment (AgustinBettati) [inline internal/service/alertconfiguration/resource_alert_configuration_test.go:426]: Will likely be able to add this test to the CI once I can use the `third_party_integration` resource and get its id directly.

File: website/docs/d/third_party_integration.markdown [hunk 3]
```diff
@@ -25,6 +25,7 @@ resource "mongodbatlas_third_party_integration" "test_datadog" {
 
 data "mongodbatlas_third_party_integration" "test" {
 	project_id = mongodbatlas_third_party_integration.test_datadog.project_id
+  type = "DATADOG"
```
Comment (AgustinBettati) [inline website/docs/d/third_party_integration.markdown:28]: small unrelated docs fix in third_party_integration data source

File: website/docs/d/alert_configuration.html.markdown [hunk 4]
```diff
@@ -226,6 +226,7 @@ Notifications to send when an alert condition is detected.
     - `WEBHOOK`
     - `MICROSOFT_TEAMS`
 
+* `integration_id` - The id of the associated integration, the credentials of which to use for requests.
```
Comment (pierwill) [inline website/docs/d/alert_configuration.html.markdown]:
```suggestion
* `integration_id` - The ID of the associated integration, the credentials of which to use for requests.
```

**Review by pierwill — APPROVED** [review]

**Review by jwilliams-mongo — APPROVED** [review]

File: website/docs/d/alert_configurations.html.markdown [hunk 5]
```diff
@@ -170,6 +170,7 @@ Notifications to send when an alert condition is detected.
     - `WEBHOOK`
     - `MICROSOFT_TEAMS`
 
+* `integration_id` - The id of the associated integration, the credentials of which to use for requests.
```
Comment (Zuhairahmed) [inline website/docs/d/alert_configurations.html.markdown]:
```suggestion
* `integration_id` - The ID of the associated integration, the credentials of which to use for requests.
```

File: website/docs/d/alert_configurations.html.markdown [hunk 6]
```diff
@@ -170,6 +170,7 @@ Notifications to send when an alert condition is detected.
...
 * `notifier_id` - The notifier id is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
```
Comment (Zuhairahmed) [inline website/docs/d/alert_configurations.html.markdown]:
```suggestion
* `notifier_id` - The notifier ID is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
```

File: website/docs/d/alert_configuration.html.markdown [hunk 7]
```diff
@@ -226,6 +226,7 @@ Notifications to send when an alert condition is detected.
...
 * `notifier_id` - The notifier id is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
```
Comment (Zuhairahmed) [inline website/docs/d/alert_configuration.html.markdown]:
```suggestion
* `notifier_id` - The notifier ID is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
```

File: website/docs/r/alert_configuration.html.markdown [hunk 8]
```diff
@@ -223,6 +239,7 @@ List of notifications to send when an alert condition is detected.
     - `WEBHOOK`
     - `MICROSOFT_TEAMS`
 
+* `integration_id` - The id of the associated integration, the credentials of which to use for requests.
```
Comment (Zuhairahmed) [inline website/docs/r/alert_configuration.html.markdown]:
```suggestion
* `integration_id` - The ID of the associated integration, the credentials of which to use for requests.
```

File: website/docs/r/alert_configuration.html.markdown [hunk 9]
```diff
@@ -223,6 +239,7 @@ List of notifications to send when an alert condition is detected.
...
 * `notifier_id` - The notifier id is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
```
Comment (Zuhairahmed) [inline website/docs/r/alert_configuration.html.markdown]:
```suggestion
* `notifier_id` - The notifier ID is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
```

**Review by Zuhairahmed — APPROVED** [review]
few small nits, otherwise LGTM

File: internal/service/alertconfiguration/resource_alert_configuration_test.go (hunk 2 above)
Comment (lantoli) [reply internal/service/alertconfiguration/resource_alert_configuration_test.go:426]: nice!

File: internal/service/alertconfiguration/resource_alert_configuration_test.go [hunk 10]
```diff
@@ -421,6 +422,32 @@ func TestAccConfigRSAlertConfiguration_withPagerDuty(t *testing.T) {
...
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckPagerDutyIntegrationID(t); acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6FactoriesWithProxy(proxyPort),
+		CheckDestroy:             checkDestroyUsingProxy(proxyPort),
+		Steps: []resource.TestStep{
+			{
+				Config: configWithPagerDutyIntegrationID(projectID, pagerDutyIntegrationID),
+				Check: resource.ComposeTestCheckFunc(
+					checkExistsUsingProxy(proxyPort, resourceName),
+					resource.TestCheckResourceAttr(resourceName, "notification.0.integration_id", pagerDutyIntegrationID),
```
Comment (lantoli) [inline internal/service/alertconfiguration/resource_alert_configuration_test.go:443]: when you enable the test, be aware that plural ds shouldn't assume any result order as it's happening here (if using  ProjectIDExecution)

File: website/docs/r/alert_configuration.html.markdown [hunk 11]
```diff
@@ -223,7 +239,8 @@ List of notifications to send when an alert condition is detected.
     - `WEBHOOK`
     - `MICROSOFT_TEAMS`
 
-* `notifier_id` - The notifier id is a system-generated unique identifier assigned to each notification method. This is needed when updating third-party notifications without requiring explicit authentication credentials.
+* `integration_id` - The ID of the associated integration, the credentials of which to use for requests.
```
Comment (lantoli) [inline website/docs/r/alert_configuration.html.markdown:242]: it's talking about credentials, is this a sensitive attribute?

**Review by lantoli — APPROVED** [review]

File: website/docs/r/alert_configuration.html.markdown (hunk 11 above)
Comment (AgustinBettati) [reply website/docs/r/alert_configuration.html.markdown:242]: The id itself is not sensitive, is is being returned by the Atlas API. The id is used to reference an existing third party integration which configures credentials.

Comment (lantoli) [reply website/docs/r/alert_configuration.html.markdown:242]: ok, i would probably remove this part (or i don't understand it maybe): "the credentials of which to use for requests."

Comment (AgustinBettati) [reply website/docs/r/alert_configuration.html.markdown:242]: I believe it is okay as it explains it uses credentials from the associated integration for requests. As this is how it is currently documented in the API docs prefer to leave as is.
